File: Python_Solutions/Arrays and Strings/URLify.py
import unittest

def urlify(url):

	if url[-1].isspace():
		url = url.rstrip() + "%20"

	urlList = list(url)

	for i in range(len(url)):
		if urlList[i] == " ":
			urlList[i] = "%20"
	return "".join(urlList)

class Test(unittest.TestCase):

	dataT = [(('ronan byrne  '),('ronan%20byrne%20')), (('ronanbyrne '),('ronanbyrne%20'))]

	def test_urlify(self):
		for [test_string, expected] in self.dataT:
			actual = urlify(test_string)
			self.assertEqual(actual, expected)

if __name__ == '__main__':
	unittest.main()

File: Java_Solutions/src/strings_and_arrays/URLify.java
package strings_and_arrays;

public class URLify {
	
	public static void main(String [] args) {
		
		urlify("Ronan Byrne");
		urlify("Ronan Byrne ");
		
	}
	
	public static String urlify(String s) {
		
		int whiteSpaceCount = 0;
		
		for (int i = 0; i < s.length(); i++) {
			if(s.charAt(i) == ' ')
				whiteSpaceCount += 3;
		}
		
		char [] changedCharList = new char [whiteSpaceCount + s.length()];
		
		int index = 0;
		for (int i = 0; i < s.length(); i++) {
			
			if (s.charAt(i) != ' ') {
				changedCharList[index] = s.charAt(i);
				index ++;
			} else {
				changedCharList[index] = '%';
				changedCharList[index+1] = '2';
				changedCharList[index+2] = '0';
				index = index +3;
			}	
				
		}
		
		String URL = new String(changedCharList);
		
		System.out.println(URL);
		
		return URL;
	}
	

}

